### ISSUE TITLE
SliverAppbar does not expand when pinned is false and using ios accessibility focus traversal

### ISSUE THREAD

**Issue by guinardA** [flutter/flutter#94096]
When you have an app with a SliverAppBar with pinned at false and a SliverList body, and you're using voiceover on iOS, scrolling to the bottom of the list and then back up will get the voice over never come back on silverAppBar.

This problem has already been reported and corrected when sliverApp has pinned true:
[issue 60331](https://github.com/flutter/flutter/issues/60331)
[issue 61360](https://github.com/flutter/flutter/issues/61360)

[Fix](https://github.com/flutter/engine/pull/19788)

But the problem persists with pinned to false

### Steps to Reproduce
Using the following example app:

```
import 'package:flutter/material.dart';

void main() {
  runApp(Demo());
}

class Demo extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: CustomScrollView(
          slivers: [
            SliverAppBar(
              title: 'title'
            ),
            SliverList(
              delegate: SliverChildListDelegate([
                for (var i = 0; i < 50; i ++)
                  Text('This is text. ' * 45),
              ]),
            ),
          ],
        ),
      ),
    );
  }
}
```

1. Open the above app with voiceover enabled on an iOS device.
2. Swipe to the right to navigate to the very bottom of the list, until the header disappears
3. Swipe back up to the header, while it is still collapsed.
4. Notice that the voice over tool gets stuck in the status bar, even when you swipe left and right, you can not get back to the page.

**Expected results:**
Swiping left to navigate back up the list will put you back in AppBar.

**Actual results:**
Voice Over gets stuck in the status bar, back button, nor text in list, can not be navigated to.

**Flutter doctor**

```
[✓] Flutter (Channel stable, 2.5.1, on macOS 11.6 20G165 darwin-x64, locale fr-FR)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.2)
[✓] VS Code (version 1.62.3)

• No issues found!
```

**Comment by mamuseferha** [comment]
Hi @guinardA I tested this on the latest stable and master channel and it is reproducible

<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel master, 2.6.0-12.0.pre.826, on macOS 11.6.1 20G224 darwin-x64, locale en-US)
    • Flutter version 2.6.0-12.0.pre.826 at /Users/flo/fvm/versions/master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6c1455a23a (5 hours ago), 2021-11-23 21:13:05 -0500
    • Engine revision 539239e256
    • Dart version 2.16.0 (build 2.16.0-42.0.dev)
    • DevTools version 2.8.0

[!] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/flo/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.62.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] Connected device (6 available)
    • SM G975U (mobile)                  • R58M95MA95E                          • android-arm64  • Android 11 (API 30)
    • Android SDK built for x86 (mobile) • emulator-5554                        • android-x86    • Android 11 (API 30) (emulator)
    • iPhone 13 (mobile)                 • EF1251D3-C9C4-475E-9F91-58CD010FDA48 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-0 (simulator)
    • iPhone (mobile)                    • 2cb2f234cda6637b9ad960bd3eae4f38fb3a5896 • ios            • iOS 15.0.2 19A404
    • macOS (desktop)                    • macos                                • darwin-x64     • macOS 11.6.1 20G224 darwin-x64
    • Chrome (web)                       • chrome                               • web-javascript • Google Chrome 93.0.4577.63

! Doctor found issues in 1 category.
```
```bash
[✓] Flutter (Channel master, 2.6.0-12.0.pre.826, on macOS 11.6.1 20G224 darwin-x64, locale en-US)
    • Flutter version 2.6.0-12.0.pre.826 at /Users/flo/fvm/versions/master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6c1455a23a (5 hours ago), 2021-11-23 21:13:05 -0500
    • Engine revision 539239e256
    • Dart version 2.16.0 (build 2.16.0-42.0.dev)
    • DevTools version 2.8.0

[!] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/flo/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.62.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] Connected device (6 available)
    • SM G975U (mobile)                  • R58M95MA95E                          • android-arm64  • Android 11 (API 30)
    • Android SDK built for x86 (mobile) • emulator-5554                        • android-x86    • Android 11 (API 30) (emulator)
    • iPhone 13 (mobile)                 • EF1251D3-C9C4-475E-9F91-58CD010FDA48 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-0 (simulator)
    • iPhone (mobile)                    • 2cb2f234cda6637b9ad960bd3eae4f38fb3a5896 • ios            • iOS 15.0.2 19A404
    • macOS (desktop)                    • macos                                • darwin-x64     • macOS 11.6.1 20G224 darwin-x64
    • Chrome (web)                       • chrome                               • web-javascript • Google Chrome 93.0.4577.63

! Doctor found issues in 1 category.
```

</details>

Thank you

**Comment by cyanglaz** [comment]
Maybe related: https://github.com/flutter/flutter/issues/92853

**Comment by cyanglaz** [comment]
cc @chunhtai

**Comment by chunhtai** [comment]
I took a look at the iOS app, it looks like they always build a pinned header in case voice over is on, they also makes sure the pinned header has all the necessary control (See the Settings app). @LongCatIsLooong  I thought it should call the show on screen action on the header if it is not on the screen?

**Comment by LongCatIsLooong** [comment]
It's been a while since I worked on that. I remember `RenderObject.shownOnScreen` is called by the namesake `SemanticsAction`? But `showOnScreen` doesn't work very well if something does not scroll linearly since `RenderViewportBase.showInViewport` makes the assumption that everything moves linearly.

**Comment by chunhtai** [comment]
>>  doesn't work very well if something does not scroll linearly since RenderViewportBase.showInViewport makes the assumption that everything moves linearly.

This should be a 1-D scroll, it should be linear?

**Comment by chunhtai** [comment]
Closing this issue as this matches native behavior
